application/associations.py

```diff
@@ -263,7 +263,10 @@ def edit_file(self, selection):
 			self._application.create_terminal_tab(active_object._notebook, options)
 
 		else:
-			subprocess.Popen(split_command, cwd=os.path.dirname(selection[0]))
+			cwd = None
```

Comment:
Right - but maybe (if it is local) we shouldn't do the file:// bit anyway - just because an app supports URIs doesn't mean we have to pass an URI as parameter instead of path, does it?

Anyway - provider would have to be passed around...